libs/ml/examples/bert_save_load_pretrained_model_example/main.cpp

```diff
@@ -0,0 +1,527 @@
+        "Key_Transform_TimeDistributed_FullyConnected_Weights",
+        "Key_Transform_TimeDistributed_FullyConnected_Bias",
+        "Value_Transform_TimeDistributed_FullyConnected_Weights",
+        "Value_Transform_TimeDistributed_FullyConnected_Bias",
+        "SelfAttentionEncoder_Multihead_Attention_MultiheadAttention_Head_No");
+  }
+
+  return std::make_pair(std::vector<std::string>({segment, position, tokens, mask}),
+                        encoder_outputs);
+}
+
+TensorType RunPseudoForwardPass(std::vector<std::string> input_nodes, std::string output_node,
+                                BERTConfig const &config, GraphType g, SizeType batch_size,
+                                bool verbose)
+{
+  std::string segment      = input_nodes[0];
+  std::string position     = input_nodes[1];
+  std::string tokens       = input_nodes[2];
+  std::string mask         = input_nodes[3];
+  std::string layer_output = output_node;
+
+  SizeType max_seq_len = config.max_seq_len;
+  SizeType seq_len     = 256u;
+
+  TensorType tokens_data({max_seq_len, batch_size});
+  tokens_data.Fill(static_cast<DataType>(1));
+
+  TensorType mask_data({max_seq_len, 1, batch_size});
+  for (SizeType i = 0; i < seq_len; i++)
+  {
+    for (SizeType b = 0; b < batch_size; b++)
+    {
+      mask_data.Set(i, 0, b, static_cast<DataType>(1));
+    }
+  }
+  TensorType position_data({max_seq_len, batch_size});
+  for (SizeType i = 0; i < seq_len; i++)
+  {
+    for (SizeType b = 0; b < batch_size; b++)
+    {
+      position_data.Set(i, b, static_cast<DataType>(i));
+    }
+  }
+  TensorType segment_data({max_seq_len, batch_size});
+  g.SetInput(segment, segment_data);
+  g.SetInput(position, position_data);
+  g.SetInput(tokens, tokens_data);
+  g.SetInput(mask, mask_data);
+
+  std::cout << "Starting forward passing on " << batch_size << " batches." << std::endl;
+  auto cur_time  = std::chrono::high_resolution_clock::now();
+  auto output    = g.Evaluate(layer_output, false);
+  auto time_span = std::chrono::duration_cast<std::chrono::duration<double>>(
+      std::chrono::high_resolution_clock::now() - cur_time);
+  std::cout << "time span: " << static_cast<double>(time_span.count()) << std::endl;
+  std::cout << "time span per batch: "
+            << static_cast<double>(time_span.count()) / static_cast<double>(batch_size)
+            << std::endl;
+  if (verbose)
+  {
+    for (std::size_t i = 0; i < output.shape().size(); i++)
+    {
+      std::cout << " | " << output.shape(i);
+    }
+    std::cout << "\nfirst token: \n"
+              << output.View(0).Copy().View(0).Copy().ToString() << std::endl;
+  }
+  return output;
+}
+
+void SaveGraphToFile(GraphType &g, std::string const file_name)
+{
+  // start serializing and writing to file
+  fetch::ml::GraphSaveableParams<TensorType> gsp1 = g.GetGraphSaveableParams();
+  std::cout << "got saveable params" << std::endl;
+
+  fetch::serializers::SizeCounter       counter;
```

Comment:
rework this to use the LargeObjectSerializeHelper in core/serializers/main_serializers.hpp

Reply:
it seems that the largeobject helper does not give me access to its bytearray data like normal serializer. it can be tricky to use I m afraid